### PR TITLE
feat: adding mutation api for metadata. cleaning up clippy complaints

### LINKED ISSUE #209: API for mutating Metadata
Ideally something like

```rs
impl Metadata {
    // ...

    pub fn put(&mut self, key: &str, value: Ipld);
}
```

### PR DESCRIPTION
**Pull request by laudiacay**
## Summary

Adds metadata mutation API. Closes #209 

This PR fixes/implements the following **bugs/features**

- [x] metadata mutation API (put, delete, update)
- [x] Clippy complaints about unnecessary async/awaits in link.rs

Motivation: necessary in order to add a pre-encryption file checksum field to metadata for file deduplication across multiple uploads in the same PrivateForest

## Test plan (required)

I added doctests for each of these and they pass.

## Closing issues

Closes #209 

## After Merge

- [ ] Does this change invalidate any docs or tutorials? _If so ensure the changes needed are either made or recorded_ (no)
- [ ] Does this change require a release to be made? Is so please create and deploy the release (no)

### CODE REVIEW

File: wnfs/src/common/link.rs
Comment (matheus23) [inline wnfs/src/common/link.rs:51]: Yeah weird I swear we fixed these in some other PR recently.
Hopefully nothing went wrong in some merge commit previously :eyes:

Comment (matheus23) [reply wnfs/src/common/link.rs:51]: Ah nevermind. This is fixed in `main`, the PR just needs to be rebased :)

Comment (laudiacay) [reply wnfs/src/common/link.rs:51]: ill rebase

Comment (laudiacay) [reply wnfs/src/common/link.rs]: ok checkem

File: wnfs-common/src/metadata.rs
Comment (matheus23) [inline wnfs-common/src/metadata.rs:151]:
```suggestion
    /// use wnfs_common::Metadata;
```

**Review by matheus23 — APPROVED** [review]
Good to go, once the imports are fixed :rocket: 

:shipit:

**Comment by laudiacay** [comment]
done! @matheus23 i'll let you merge

**Pull request merged**
